CraftBukkit/0161-Add-PlayerSettingsUpdateEvent.patch

```diff
@@ -0,0 +1,298 @@
+From 6af904d2a94beedfb6498870cf593030be2c04cc Mon Sep 17 00:00:00 2001
+From: Electroid <electroidfilms@gmail.com>
+Date: Sat, 17 Sep 2016 13:31:16 -0700
+Subject: [PATCH] Add PlayerSettingsUpdateEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
+index 5da5a49..374c081 100644
+--- a/src/main/java/net/minecraft/server/EntityPlayer.java
++++ b/src/main/java/net/minecraft/server/EntityPlayer.java
+@@ -23,23 +23,20 @@ import org.bukkit.craftbukkit.entity.CraftPlayer;
+ import org.bukkit.craftbukkit.event.CraftEventFactory;
+ import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.event.inventory.InventoryType;
+-import org.bukkit.event.player.PlayerChangedMainHandEvent;
+ import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
+-import org.bukkit.inventory.MainHand;
+ // CraftBukkit end
+ 
+ // SportBukkit start
+-import org.bukkit.craftbukkit.util.Skins;
+-import org.bukkit.event.player.PlayerSkinPartsChangeEvent;
++import org.bukkit.PlayerSettings;
+ import org.bukkit.event.player.PlayerKnockbackEvent;
+ import org.bukkit.event.player.PlayerVelocityEvent;
++import org.bukkit.util.EnumUtils;
+ import org.bukkit.util.Vector;
+ // SportBukkit end
+ 
+ public class EntityPlayer extends EntityHuman implements ICrafting {
+ 
+     private static final Logger bS = LogManager.getLogger();
+-    public String locale = null; // SportBukkit - make public and default to null
```

Comment:
Could we copy to/from these NMS fields instead of replacing them? This approach could make rebasing difficult.